bin/run_atarchiver.sh

```diff
@@ -2,44 +2,54 @@
 loc="`dirname \"$0\"`"
 source $loc/base_init.sh
 
-if [ -z $1 ];
-then
-    echo "$0: missing argument: lsst_dds_partition_prefix container_version"
-    exit 1
-fi
+error_msg="$0: missing argument: -p [summit|ncsa] -c container_version"
+
+export CONTAINER_VERSION=
+export ARCHIVE_SITE=
+export LSST_DDS_INTERFACE=
 
-if [ -z $2 ];
-then
-    echo "$0: missing argument: lsst_dds_partition_prefix container_version"
+while getopts p:c: option
+do
+case "${option}"
+in
+p)
+    if [ "$OPTARG" = "summit" ]; then
+        ARCHIVE_SITE=$OPTARG
+        LSST_DDS_INTERFACE=em2.1201
+    elif [ "$OPTARG" = "ncsa" ]; then
+        ARCHIVE_SITE=$OPTARG
+        LSST_DDS_INTERFACE=p3p2
+    else
+        echo "argument must be summit or ncsa"
+        exit 1
+    fi;;
+c) CONTAINER_VERSION=${OPTARG};;
+esac
+done
+
+if [[ -z $LSST_DDS_INTERFACE ]] || [[ -z $CONTAINER_VERSION ]]; then
+    echo $error_msg
```

Comment:
This check tells me that neither of the script's options is really an option. Their values are effectively mandatory arguments and I don't quite understand what prevents you from treating them as such.

Reply:
I'm not sure what you're asking - both arguments are mandatory.   This is one way in bash to check to see all arguments are set.

If they were optional, the error syntax would be something like "script [-c foo] [-p value1 | value 2]".

Reply:
My point was essentially that using syntax like `script -p value1 -c value2` (and using `getopts` internally) where `value1` and `value2` are mandatory arguments (instead `script value1 value2`) violates, in my opinion, the principle of the least surprise.

Though I'm sure there are exceptions, most of the time it is quite safe to assume (when using a command line tool) that anything what starts with a dash is an option. I'm not saying that what you're doing is wrong. I was just curios what made you go against a pretty common convention.

Reply:
I specified it that way to make it clear which option was which, and to make sure that required values were used (summit, NCSA).  Trying to make this user proof.